sitestats/sitestats-impl/src/java/org/sakaiproject/sitestats/impl/event/detailed/DetailedEventsManagerImpl.java

```diff
@@ -375,4 +376,33 @@ public List<String> getUsersForTracking(String siteID)
 			return Collections.emptyList();
 		}
 	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	@Override
+	public long getDetailedEventsCount(final TrackingParams trackingParams)
+	{
+		if (!statMan.isDisplayDetailedEvents() || !statsAuthz.canCurrentUserTrackInSite(trackingParams.siteId))
+		{
+			return 0;
+		}
+
+		HibernateCallback<Long> hcb = session ->
+		{
+			Optional<Criteria> critOpt = basicCriteriaForTrackingParams(session, trackingParams);
+			if (!critOpt.isPresent())
+			{
+				return 0L;
+			}
+			
+			Criteria crit = critOpt.get();
+			// Use the Hibernate rowCount projection to get the total count
+			crit.setProjection(Projections.rowCount());
+			
+			return (Long) crit.uniqueResult();
```

Comment:
this is exactly the on place where a uniqueResult should be used when performing a row count! 👍 

This is a little more resilient as it doesn't append on the type that is returned
```suggestion
			return ((Number) crit.uniqueResult()).longValue();
```